### PR TITLE
Add extra code coverage tests

### LINKED ISSUE #28694: [Testerina] Code coverage fails with nio.AccessDeniedException when having extra tests
**Description:**
When adding extra tests to the code coverage, the windows build fails with the following error

```
testerina-intg-test-suite > testerina-intg-tests > org.ballerinalang.testerina.test.TestReportTest > testWithCoverage STANDARD_ERROR
    Feb 17, 2021 5:37:31 AM org.ballerinalang.test.context.ServerLogReader feedAndPrint
    SEVERE: error: unable to clean the target directory: Unable to delete directory C:\Users\RUNNER~1\AppData\Local\Temp\bal-test-integration-testerina-project-17320448599123816210\project-based-tests\test-report-tests\target\report.

testerina-intg-test-suite > testerina-intg-tests > org.ballerinalang.testerina.test.TestReportTest > testWithCoverage FAILED
    java.nio.file.AccessDeniedException: C:\Users\RUNNER~1\AppData\Local\Temp\bal-test-integration-testerina-project-17320448599123816210\project-based-tests\test-report-tests\target\report\test_results.json
        at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:89)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
        at java.base/sun.nio.fs.WindowsFileSystemProvider.newByteChannel(WindowsFileSystemProvider.java:235)
        at java.base/java.nio.file.Files.newByteChannel(Files.java:371)
        at java.base/java.nio.file.Files.newByteChannel(Files.java:422)
        at java.base/java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:420)
        at java.base/java.nio.file.Files.newInputStream(Files.java:156)
        at java.base/java.nio.file.Files.newBufferedReader(Files.java:2839)
        at org.ballerinalang.testerina.test.TestReportTest.runCommand(TestReportTest.java:129)
        at org.ballerinalang.testerina.test.TestReportTest.testWithCoverage(TestReportTest.java:77)
        
```

**Steps to reproduce:**
Add the following changes removed by the commit : https://github.com/ballerina-platform/ballerina-lang/pull/28236/commits/756304ba029297ebffd19009934fa12e5d91b4ca

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by Ibaqu**
## Purpose
Adds code coverage tests for module wise execution

Fixes #28694

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/TestReportTest.java [hunk 1]
```diff
@@ -46,15 +46,20 @@
...
+    private static final int SKIPPED = 2;
+    private static final int FAILED = 3;
+
     @BeforeClass
     public void setup() throws BallerinaTestException {
         balClient = new BMainInstance(balServer);
         projectPath = projectBasedTestsPath.resolve("test-report-tests").toString();
-        resultsJsonPath = projectBasedTestsPath.resolve("test-report-tests").resolve("target").resolve("report")
-                .resolve("test_results.json");
+        resultsJsonPath = projectBasedTestsPath.resolve("test-report-tests").resolve("target").resolve(
```
Comment (hemikak) [inline tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/TestReportTest.java:58]: Break the new line from the start of `.resolve(`

Comment (Ibaqu) [reply tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/TestReportTest.java]: Corrected formatting

**Review by hemikak — APPROVED** [review]

File: tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/TestReportTest.java [hunk 2]
```diff
@@ -76,70 +81,111 @@ public void testWarningForJacocoXMLFlag() throws BallerinaTestException {
...
+    }
+
+    private void runCommand(String[] args) throws BallerinaTestException {
+        balClient.runMain("test", args, null, new String[]{}, new LogLeecher[]{}, projectPath);
+
         Gson gson = new Gson();
-        BufferedReader bufferedReader = Files.newBufferedReader(resultsJsonPath, StandardCharsets.UTF_8);
-        resultObj = gson.fromJson(bufferedReader, JsonObject.class);
+
+        try {
```
Comment (Dilhasha) [inline tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/TestReportTest.java:141]: Let's use try with resources here.

`try(BufferedReader bufferedReader = Files.newBufferedReader(resultsJsonPath, StandardCharsets.UTF_8)){`

File: tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/TestReportTest.java [hunk 3]
```diff
@@ -255,5 +413,6 @@ private void validateCoverage() {
         Assert.assertEquals(totalCovered, resultObj.get("coveredLines").getAsInt());
         Assert.assertEquals(totalMissed, resultObj.get("missedLines").getAsInt());
         Assert.assertEquals(coveragePercentage, resultObj.get("coveragePercentage").getAsFloat());
+
```
Comment (Dilhasha) [inline tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/TestReportTest.java:416]: unnecessary new line.

File: tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/TestReportTest.java (hunk 2 above)
Comment (Ibaqu) [reply tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/TestReportTest.java]: Made use of `try-with-resources` for the buffered reader

File: tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/TestReportTest.java (hunk 3 above)
Comment (Ibaqu) [reply tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/TestReportTest.java]: Removed NL

**Review by Dilhasha — APPROVED** [review]

**Pull request merged**